feat: ui/ux improvements DHIS2-10143

edoardo opened the pull request:
Implements [DHIS2-10143](https://jira.dhis2.org/browse/DHIS2-10143) (see description below)

**Requires https://github.com/dhis2/analytics/pull/894**

---

### Key features

1. fixes [DHIS2-10147](https://jira.dhis2.org/browse/DHIS2-10147)
2. fixes [DHIS2-10153](https://jira.dhis2.org/browse/DHIS2-10153)
3. fixes [DHIS2-10154](https://jira.dhis2.org/browse/DHIS2-10154)
4. fixes [DHIS2-10150](https://jira.dhis2.org/browse/DHIS2-10150)
5. fixes [DHIS2-10146](https://jira.dhis2.org/browse/DHIS2-10146)
6. fixes [DHIS2-10151](https://jira.dhis2.org/browse/DHIS2-10151)
7. fixes [DHIS2-10148](https://jira.dhis2.org/browse/DHIS2-10148)
8. fixes [DHIS2-10149](https://jira.dhis2.org/browse/DHIS2-10149)
9. fixes [DHIS2-10144](https://jira.dhis2.org/browse/DHIS2-10144)
10. fixes [DHIS2-10152](https://jira.dhis2.org/browse/DHIS2-10152)
11. fixes [DHIS2-10145](https://jira.dhis2.org/browse/DHIS2-10145)
12. remove MUI dependencies

---

### TODO

- [x] update `analytics` dependency

---

### Description

The initial idea was to replace all MUI components and icons with equivalents from `ui`.
Then @cooper-joe brought to my attention [this ticket](https://jira.dhis2.org/browse/DHIS2-10143) which covers all ui/ux improvements and upgrades for dashboards.
So I ended up implementing the new design at the same time.
A few places have something different than before, but mostly is the same just with new icons.

1. [Use ui components for Add Filter button](https://jira.dhis2.org/browse/DHIS2-10147)
before:
![Screenshot 2021-04-16 at 10 41 58](https://user-images.githubusercontent.com/150978/114997701-6b794e00-9ea0-11eb-9ca3-642cc48e4d2f.png)
after:
![Screenshot 2021-04-19 at 16 36 27](https://user-images.githubusercontent.com/150978/115254250-74cb1a80-a12d-11eb-8efe-7edfbb45f418.png)

2. [Use ui components for Dashboard Item Delete](https://jira.dhis2.org/browse/DHIS2-10153)
before:
![Screenshot 2021-04-16 at 10 44 39](https://user-images.githubusercontent.com/150978/114998016-cd39b800-9ea0-11eb-9faf-6f971f3a2bd0.png)
after:
![Screenshot 2021-04-16 at 11 00 27](https://user-images.githubusercontent.com/150978/115000388-02dfa080-9ea3-11eb-8c2b-24af03aa83d9.png)

3. [Use ui-icons in Dashboard "Add items" menu](https://jira.dhis2.org/browse/DHIS2-10154)
before:
![Screenshot 2021-04-16 at 10 48 09](https://user-images.githubusercontent.com/150978/114998467-45a07900-9ea1-11eb-948c-15daf2c42a89.png)
after:
![Screenshot 2021-04-16 at 10 48 15](https://user-images.githubusercontent.com/150978/114998480-49cc9680-9ea1-11eb-94d5-6f59ab133b1b.png)

4. [Update dashboard item "More" button](https://jira.dhis2.org/browse/DHIS2-10150)
before:
![Screenshot 2021-04-16 at 10 49 49](https://user-images.githubusercontent.com/150978/114998700-81d3d980-9ea1-11eb-9c47-2a61a5d606e7.png)
after:
![Screenshot 2021-04-16 at 10 49 57](https://user-images.githubusercontent.com/150978/114998726-85676080-9ea1-11eb-9fd8-6c674978f45c.png)

5. [Use ui-icons in Dashboard Title](https://jira.dhis2.org/browse/DHIS2-10146)
before:
![Screenshot 2021-04-16 at 10 51 36](https://user-images.githubusercontent.com/150978/114999002-c1022a80-9ea1-11eb-9f89-dec217290f2c.png)
after:
![Screenshot 2021-04-16 at 10 51 43](https://user-images.githubusercontent.com/150978/114999027-c52e4800-9ea1-11eb-8e2a-c71ad337c7e5.png)

6. [Update icons and labels in dashboard item menu](https://jira.dhis2.org/browse/DHIS2-10151)
before:
![Screenshot 2021-04-16 at 10 52 55](https://user-images.githubusercontent.com/150978/114999223-ee4ed880-9ea1-11eb-8295-654ca682d454.png)
after:
![Screenshot 2021-04-16 at 10 53 00](https://user-images.githubusercontent.com/150978/114999234-f1e25f80-9ea1-11eb-8c33-c0d83b758436.png)

7. [Use ui components for More button](https://jira.dhis2.org/browse/DHIS2-10148)
before:
![Screenshot 2021-04-16 at 10 54 06](https://user-images.githubusercontent.com/150978/114999531-3cfc7280-9ea2-11eb-96c2-506cef9ad25c.png)
after:
![Screenshot 2021-04-19 at 16 36 32](https://user-images.githubusercontent.com/150978/115254294-7c8abf00-a12d-11eb-918b-9d41ab7dae1b.png)

8. [Use ui-icons in search input](https://jira.dhis2.org/browse/DHIS2-10149)
before:
![Screenshot 2021-04-16 at 14 04 56](https://user-images.githubusercontent.com/150978/115021811-c325b280-9ebc-11eb-9ec1-86ff4fafbf3a.png)
after:
![Screenshot 2021-04-16 at 14 00 49](https://user-images.githubusercontent.com/150978/115021384-30851380-9ebc-11eb-96d1-4f0b57f48e49.png)

9. [Replace circle "Create dashboard" button with ui button](https://jira.dhis2.org/browse/DHIS2-10144)
before:
![Screenshot 2021-04-16 at 14 01 47](https://user-images.githubusercontent.com/150978/115021488-4f83a580-9ebc-11eb-939e-e13d0ba180bf.png)
after:
![Screenshot 2021-04-16 at 14 01 52](https://user-images.githubusercontent.com/150978/115021503-54485980-9ebc-11eb-9dcb-89ab2307f092.png)

10. [Use ui-icons in Exit Print Preview button](https://jira.dhis2.org/browse/DHIS2-10152)
before:
![Screenshot 2021-04-16 at 14 02 59](https://user-images.githubusercontent.com/150978/115021631-7f32ad80-9ebc-11eb-9a36-4c8bc330d67a.png)
after:
![Screenshot 2021-04-16 at 14 03 06](https://user-images.githubusercontent.com/150978/115021647-835ecb00-9ebc-11eb-9d25-432a19e3490a.png)

11. [Use ui-icons in dashboard chip](https://jira.dhis2.org/browse/DHIS2-10145)
before:
![Screenshot 2021-04-16 at 14 04 05](https://user-images.githubusercontent.com/150978/115021736-a5584d80-9ebc-11eb-9600-48aed69b8418.png)
after:
![Screenshot 2021-04-16 at 14 04 17](https://user-images.githubusercontent.com/150978/115021750-aab59800-9ebc-11eb-98c1-3bbaa22b45bf.png)

edoardo committed:
refactor: replace MUI icons with ui icons

edoardo committed:
fix: fix usage of icons from analytics

Requires updated analytics

edoardo committed:
refactor: use ui-icons, other ui fixes DHIS2-10151

edoardo committed:
fix: fix margin DHIS2-10150

edoardo committed:
refactor: use ui components and icons DHIS2-10148

edoardo committed:
refactor: use ui components and icons DHIS2-10153

edoardo committed:
refactor: replace more MUI icons

edoardo committed:
refactor: use ui components and icons DHIS2-10146

edoardo committed:
refactor: use ui-icons

edoardo committed:
refactor: remove unnecessary code

edoardo committed:
refactor: use ui components and icons DHIS2-10147

edoardo committed:
refactor: replace MUI with ui components

edoardo committed:
refactor: remove MUI theme and related code

edoardo committed:
chore: remove MUI dependencies

edoardo committed:
test: fix text match after changes in component

edoardo committed:
test: update test snapshots

edoardo committed:
test: initialise the mock store correctly

edoardo committed:
refactor: use icon from ui-icons DHIS2-10145

edoardo committed:
refactor: use icon from ui-icons DHIS2-10152

edoardo committed:
refactor: icons from ui-icons and UI updates DHIS2-10144

edoardo committed:
refactor: use search icon from ui-icons DHIS2-10149

edoardo committed:
test: fix Chip test and upgrade test snapshots

edoardo committed:
chore: update pot file

edoardo committed:
test: fix cypress tests for item context menu

edoardo committed:
test: restored lost data-test props used by cypress

edoardo committed:
refactor: implement controlled DropdownButton

This is used in both the Filter and More buttons.
The button's open state needs to be controlled by the app to avoid that
once open, it cannot be closed unless manually triggered.

edoardo committed:
refactor: use the local custom DropdownButton

edoardo committed:
refactor: use the custom DropdownButton

The double state is because otherwise the content of both small and
normal buttons are opened on top of each other.
This breaks cypress and you can see the 2 menus overlapping.

edoardo committed:
refactor: use local icons to fix failing tests

jenniferarnesen committed:
Merge branch 'master' into feat/ui-ux-improvements-DHIS2-10143

jenniferarnesen committed:
Merge branch 'master' into feat/ui-ux-improvements-DHIS2-10143

edoardo committed:
chore: update analytics dependency

cooper-joe reviewed (changes requested):
There's an issue with two of the changes, the menus that open for the "1. Add filter" and "7. More" buttons. These menus are opening aligned `bottom end`, whereas they should be aligned `bottom start` (using the positioning terms from the [Popper ui component](https://ui.dhis2.nu/demo/?path=/story/helpers-popper--bottom-start). I don't think these positioning issues can be ignored, they act differently than the platform standards.

Here's a simplified example of the `bottom start` positioning that should be used:
![image](https://user-images.githubusercontent.com/33054985/115249335-c7ee9e80-a128-11eb-95ef-a5743214a331.png)

edoardo committed:
fix: fix position of DropdownButton content

edoardo committed:
fix: fix toggling of button

edoardo commented:
> Here's a simplified example of the `bottom start` positioning that should be used:
> ![image](https://user-images.githubusercontent.com/33054985/115249335-c7ee9e80-a128-11eb-95ef-a5743214a331.png)

Fixed in the custom component, so it applies to both Filter and More buttons.
Updated screenshots.

edoardo committed:
fix: pass insertTheme to d2-ui components to fix narrow dialogs
